dpx/src/dpx_pdfobj.rs

```diff
@@ -1493,9 +1491,8 @@ pub unsafe extern "C" fn pdf_new_stream(mut flags: i32) -> *mut pdf_obj {
      */
     (*data).dict = pdf_new_dict();
     (*data)._flags = flags;
-    (*data).stream = 0 as *mut u8;
-    (*data).stream_length = 0_u32;
-    (*data).max_length = 0_u32;
+    let uninit_stream = std::mem::replace(&mut (*data).stream, Vec::new());
+    std::mem::forget(uninit_stream); // TODO: this is ugly
```

Comment:
You can just initialize it before moving it to the allocated memory.

```rust
let data = new(std::mem::size_of::<pdf_stream>() as u32) as *mut pdf_stream;
let new_stream = pdf_stream {
    dict: pdf_new_dict(),
    _flags: flags,
    stream: Vec::new(),
    ...
};
*data = new_stream;
```

Reply:
This would still drop the uninitialized `pdf_stream` struct during `*data = new_stream`.
It looks like `ptr::write` is made for this as it doesn't drop the previous content.